azure_ai_search_rag.ipynb

```diff
@@ -0,0 +1,908 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Generate Embeddings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 58,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def generate_embeddings(documents, model):\n",
+    "    for document in documents:\n",
+    "        content = document['content']\n",
+    "        embed = mistral.embeddings(\n",
+    "            model=model, input=[content]\n",
+    "        )\n",
+    "        # Add the embedding to the document\n",
+    "        document['embedding'] = embed.data[0].embedding\n",
+    "    return documents\n",
+    "\n",
+    "embed_model = \"mistral-embed\"\n",
+    "transformed_documents_with_embeddings = generate_embeddings(transformed_documents, embed_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Azure AI Search doesn't allow certain unsafe keys so we'll base64 encode `id` here"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 78,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import base64\n",
+    "\n",
+    "def encode_key(key):\n",
+    "    return base64.urlsafe_b64encode(key.encode()).decode()\n",
+    "\n",
+    "# Use the function to encode keys\n",
+    "for document in transformed_documents_with_embeddings:\n",
+    "    document['id'] = encode_key(document['id'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Upload Documents"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 82,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Uploaded batch 1 of 21\n",
+      "Uploaded batch 2 of 21\n",
+      "Uploaded batch 3 of 21\n",
+      "Uploaded batch 4 of 21\n",
+      "Uploaded batch 5 of 21\n",
+      "Uploaded batch 6 of 21\n",
+      "Uploaded batch 7 of 21\n",
+      "Uploaded batch 8 of 21\n",
+      "Uploaded batch 9 of 21\n",
+      "Uploaded batch 10 of 21\n",
+      "Uploaded batch 11 of 21\n",
+      "Uploaded batch 12 of 21\n",
+      "Uploaded batch 13 of 21\n",
+      "Uploaded batch 14 of 21\n",
+      "Uploaded batch 15 of 21\n",
+      "Uploaded batch 16 of 21\n",
+      "Uploaded batch 17 of 21\n",
+      "Uploaded batch 18 of 21\n",
+      "Uploaded batch 19 of 21\n",
+      "Uploaded batch 20 of 21\n",
+      "Uploaded 10000 documents in total\n"
+     ]
+    }
+   ],
+   "source": [
+    "from azure.identity import DefaultAzureCredential\n",
+    "from azure.core.credentials import AzureKeyCredential\n",
+    "from azure.search.documents import SearchClient\n",
+    "\n",
+    "\n",
+    "def upload_documents(index_name, endpoint, credential, documents):\n",
+    "    client = SearchClient(\n",
+    "        endpoint=endpoint, index_name=index_name, credential=credential\n",
+    "    )\n",
+    "\n",
+    "    # Split documents into chunks of 500\n",
```

Comment:
Nit: the word "chunk" is so commonly used for passage chunking that's best not to use it for other contexts to avoid confusion, maybe say in "in batches of 500"?

Reply:
I actually forgot about using SearchIndexingBufferedSender in the SDK :) Updated